File: cpSubeenProblem/cpSubeenProblem27.java
/*
 * http://cpbook.subeen.com/2013/01/sort-them-all.html
 */
package solvingcpsubeen;

/* package whatever; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */
class Ideone
{
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner scan = new Scanner(System.in);

        int terms;
        terms = scan.nextInt();

        for (int i = 0; i < terms; i++) {

            int num1 = scan.nextInt();
            int num2 = scan.nextInt();
            int num3 = scan.nextInt();
            int temp;
            int[] arr = new int[3];

            try {
                if (num1 < num2) {
                    arr[0] = num1;
                    arr[1] = num2;
                } else {
                    arr[0] = num2;
                    arr[1] = num1;
                }

                if (arr[1] < num3) {
                    arr[2] = num3;
                } else {
                    if (arr[0] > num3) {
                        temp = arr[1];
                        arr[1] = arr[0];
                        arr[2] = temp;
                        arr[0] = num3;
                    } else {
                        temp = arr[1];
                        arr[1] = num3;
                        arr[2] = temp;
                    }
                }

                System.out.println("Case " + (i + 1) + ": " + arr[0] + " " + arr[1] + " " + arr[2]);

            } catch (Exception e) {
                System.out.println(e);
            }
        }
	}
}

File: cpSubeenProblem/cpSubeenProblem_27.c
#include <stdio.h>
// http://cpbook.subeen.com/2013/01/sort-them-all.html

// --------------
//
// Solve without using array !
//
//---------------


int main(){

    int term, i;

    int firstNumber, secondNumber, thirdNumber;

    int temp;


    scanf("%d",&term);

    for(i = 0; i < term; i++){


        scanf("%d %d %d",&firstNumber, &secondNumber, &thirdNumber);


        if(firstNumber > secondNumber){

            temp = firstNumber;

            firstNumber = secondNumber;

            secondNumber = temp;
        }

        if(secondNumber > thirdNumber){

            temp = thirdNumber;

            thirdNumber = secondNumber;

            secondNumber = temp;        }

        if(firstNumber > secondNumber){

            temp = firstNumber;

            firstNumber = secondNumber;

            secondNumber = temp;
        }


        printf("Case %d: %d %d %d\n",(i+1), firstNumber, secondNumber, thirdNumber);

    }

    return 0;
}
